javatools/src/main/java/org/xvm/tool/Launcher.java

```diff
@@ -2229,6 +2237,13 @@ private File[] listFiles(File dir)
         {
         File[] aFile = dir.listFiles();
         Arrays.sort(aFile, Comparator.comparing(File::getName));
+        // Psuedo random shuffle files, in an effort to trigger bugs
+        int seed = options().seed();
+        if( seed!=0 ) System.err.println("Compiling with seed="+seed);
```

Comment:
I agree on the style. As Cliff said himself earlier, a change in an existing file should follow the current style used. Otherwise it's going to be a mess.

Reply:
You are introducing your own sorting logic for the seed, when there is sorting logic in the JDK. 

private final Random random = seed == 0 ? null : new Random(seed)

... 
if (seed != 0) {
    Collections.shuffle(dir, random)
}

Reply:
On the substance, my concern is that 

- this solution can only cover functionality in Launcher and derived classes, while there is no visibility of "seed" to the rest of the compiler/runtime
- even more importantly there is no visibility into the Ecstasy code, which I think would have to be exposed via some injection

Reply:
Also, I think using of the random would be better. The proposed algorithm doesn't really shuffle much. I wrote a simple test for it:

    module TestSimple
        {
        @Inject Console console;
    
        void run()
            {
            String[] strings = ["a", "b", "c", "d", "e", "f", "g"];
    
            console.print(shuffle(strings, 1));
            console.print(shuffle(strings, 2));
            console.print(shuffle(strings, 3));
            console.print(shuffle(strings, 4));
            console.print(shuffle(strings, 5));
            console.print(shuffle(strings, 6));
            }
    
        String[] shuffle(String[] strings, Int seed)
            {
            strings = strings.reify(Mutable);
            for (Int i : 0 ..< strings.size)
                {
                Int j = (i + seed) % strings.size;
                String tmp = strings[i]; strings[i] = strings[j]; strings[j] = tmp;
                }
            return strings;
            }
        }

Here's the output:

[a, c, d, e, f, g, b]
[b, a, e, f, g, c, d]
[b, c, a, g, d, e, f]
[d, a, b, c, f, g, e]
[c, d, e, a, b, g, f]
[b, c, d, e, f, a, g]

Reply:
Cam, what do you think of adding `List shuffle(Random random)` to List.x?